src/http-client/index.ts

```diff
@@ -0,0 +1,19 @@
+export { FetchClient } from "./fetch-client";
+
+export type HTTPRequest = {
+  data: Record<string, any>;
+  headers: Record<string, string>;
+  method: string;
+  url: string;
+  keepalive?: boolean;
+};
+
+export type HTTPResponse = {
+  headers: Record<string, string>;
+  status: number;
+  body: unknown;
```

Comment:
I suggest we add a new type - QueryResponse<T> = QuerySuccess<T> | QueryFailure; and then here do:

```suggestion
  body: QueryResponse<JSONValue>;
```

as fauna will only ever get us back JSON